spring/boot-webflux-autoconfigure/src/main/java/com/linecorp/armeria/spring/web/reactive/ArmeriaReactiveWebServerFactory.java

```diff
@@ -257,16 +260,14 @@ private void configureArmeriaService(ServerBuilder sb, ArmeriaSettings settings)
         }
     }
 
-    private <T> Optional<T> findBean(Class<T> clazz) {
-        final List<T> beans = findBeans(clazz);
-        switch (beans.size()) {
-            case 0:
-                return Optional.empty();
-            case 1:
-                return Optional.of(beans.get(0));
-            default:
-                throw new IllegalStateException("Too many " + clazz.getSimpleName() + " beans: " +
-                                                beans + " (expected: 1)");
+    @VisibleForTesting
+    <T> Optional<T> findBean(Class<T> clazz) {
+        try {
+            return Optional.of(beanFactory.getBean(clazz));
+        } catch (NoUniqueBeanDefinitionException e) {
```

Comment:
Just curious. From `getBean` javadoc:
```
@throws NoUniqueBeanDefinitionException if more than one bean of the given type was found
```
So I'm curious whether the problem you described would be resolved or not. Is this fix working when multiple `MeterRegistry` beans are specified? How about adding a test for it?

Reply:
I checked with my personal project that it is working.

I thought about adding tests for this change, it seems like testing against Spring framework, so I just gave up.

Let me think what I can do.

Reply:
Added tests to show behavioral difference.